### PR TITLE
checksumming responses

### PR DESCRIPTION
**Pull request by ashelly**
simple sum of characters since last newline (mod 256) inserted before each newline.

### CODE REVIEW

File: serial.c [hunk 1]
```diff
@@ -37,6 +37,8 @@ uint8_t tx_buffer[TX_BUFFER_SIZE];
 uint8_t tx_buffer_head = 0;
 volatile uint8_t tx_buffer_tail = 0;
 
+uint8_t checksum;  //sum all bytes between newlines.
```
Comment (zsalzbank) [inline serial.c]: Need to initialize?

Comment (ashelly) [inline serial.c]: Yes. (it works now b/c globals are automatically 0, but yes.)

**Comment by penguinland** [comment]
LGTM

**Comment by mdicicco** [comment]
Wow, so simple.

**Pull request merged**